File: COMPILADOR/kokoslan_/src/src/java/ast/KoKoAst.java
/**
 @author loriacarlos@gmail.com
 
*/
package src.java.ast;
import java.io.*;
import src.java.eval.KoKoContext;
import src.java.eval.KoKoValue;


public interface KoKoAst{
   default void genCode(){
      genCode(System.out);
   }
   default void genCode(PrintStream out){
   }
   KoKoValue eval(KoKoContext ctx);
   
}

File: COMPILADOR/kokoslan/src/KoKoKotlin/ast/KoKoAst.kt
package src.java.ast

import KoKoKotlin.eval.KoKoContext
import KoKoKotlin.eval.KoKoValue
import java.*
import java.io.*


interface KoKoAst {
    fun genCode(out: PrintStream = System.out) {
    }

    fun eval(ctx: KoKoContext): KoKoValue

}